scripts/setup.sh

```diff
@@ -20,43 +24,85 @@ if [ "$EUID" -ne 0 ]; then
   exit $?  # Exit with the status of the sudo command
 fi
 
+# --- Autologin Configuration ---
 echo -e "${INFO}Configuring autologin...${RESET}"
 if grep -q "autologin" "/etc/systemd/system/getty@tty1.service.d/autologin.conf" 2>/dev/null; then
   echo -e "${SUCCESS}\tautologin is already enabled!${RESET}."
 else
   if command -v raspi-config >/dev/null 2>&1; then
     echo -e "${DEBUG}Enabling autologin using raspi-config...${RESET}"
     raspi-config nonint do_boot_behaviour B4
+    echo -e "${SUCCESS}\tAutologin has been enabled!${RESET}"
   else
     echo -e "${ERROR}Could not enable autologin${RESET}"
     echo -e "${ERROR}Please configure autologin manually and rerun setup.${RESET}"
+    exit 1
   fi
-  echo -e "${SUCCESS}\tautologin has been enabled!${RESET}"
 fi
 
+# --- Dependency Installation ---
 echo -e "${INFO}Installing dependencies...${RESET}"
-apt install -y git jq wtype nodejs npm
+apt update && apt install -y git jq wtype curl unzip
 
+# --- Repo Cloning ---
 echo -e "${INFO}Cloning repository...${RESET}"
+rm -rf "$PIOSK_DIR"
```

Comment:
This will remove the current `config.json` without backing it up first; needs to be addressed before the release @parthsidpara